example/lib/go_router_example.dart

```diff
@@ -0,0 +1,306 @@
+        branches: <StatefulShellBranch>[
+          StatefulShellBranch(
+            navigatorKey: _sectionANavigatorKey,
+            routes: <RouteBase>[
+              GoRoute(
+                path: '/a',
+                builder:
+                    (final BuildContext context, final GoRouterState state) =>
+                        const RootScreen(label: 'A', detailsPath: '/a/details'),
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: 'details',
+                    builder: (
+                      final BuildContext context,
+                      final GoRouterState state,
+                    ) =>
+                        const DetailsScreen(label: 'A'),
+                  ),
+                ],
+              ),
+            ],
+          ),
+          StatefulShellBranch(
+            routes: <RouteBase>[
+              GoRoute(
+                path: '/b',
+                builder:
+                    (final BuildContext context, final GoRouterState state) =>
+                        const RootScreen(
+                  label: 'B',
+                  detailsPath: '/b/details/1',
+                  secondDetailsPath: '/b/details/2',
+                ),
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: 'details/:param',
+                    builder: (
+                      final BuildContext context,
+                      final GoRouterState state,
+                    ) =>
+                        DetailsScreen(
+                      label: 'B',
+                      param: state.pathParameters['param'],
+                    ),
+                  ),
+                ],
+              ),
+            ],
+          ),
+          StatefulShellBranch(
+            routes: <RouteBase>[
+              GoRoute(
+                path: '/c',
+                builder:
+                    (final BuildContext context, final GoRouterState state) =>
+                        const RootScreen(
+                  label: 'C',
+                  detailsPath: '/c/details',
+                ),
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: 'details',
+                    builder: (
+                      final BuildContext context,
+                      final GoRouterState state,
+                    ) =>
+                        DetailsScreen(
+                      label: 'C',
+                      extra: state.extra,
+                    ),
+                  ),
+                ],
+              ),
+            ],
+          ),
+        ],
+      ),
+    ],
+  );
+
+  @override
+  Widget build(final BuildContext context) => MaterialApp.router(
+        title: 'Flutter Demo',
+        theme: ThemeData(
+          colorSchemeSeed: Colors.deepPurple,
+          useMaterial3: true,
+        ),
+        routerConfig: _router,
+      );
+}
+
+class ScaffoldWithNavBar extends StatelessWidget {
```

Comment:
The name is quite misleading. It should better reflect that this is where the package comes into play. Something like ``AdaptiveNavigationShell`` would come to my mind.

Reply:
Done